Template10 (Library)/Services/NavigationService/NavigationLogic.cs

```diff
@@ -171,6 +171,10 @@ public void SetupViewModel(INavigationService service, ITemplate10ViewModel view
                 var vm = viewmodel as Portable.IConfirmNavigationAsync;
                 return !await vm?.CanNavigateAsync(parameters);
             }
+            else if (viewmodel != null)
```

Comment:
@JerryNixon But `else` is returning `true`. See the original code. We don't want 'true' returned for non ViewModel datacontext.